src/hypothesis/strategies.py

```diff
@@ -982,32 +987,61 @@ def fractions(min_value=None, max_value=None, max_denominator=None):
 
     If max_denominator is not None then the absolute value of the denominator
     of any generated values is no greater than max_denominator. Note that
-    max_denominator must be at least 1.
+    max_denominator must be None or a positive integer, and the denominator
```

Comment:
> Should not crash with AssertionError

I absolutely agree, and hence 11b88e9f688bd47209f77ace416036b526d2c869.  Regardless of how we validate though, any valid bounds to `fractions()` must internally be converted to valid bounds for `integers()`.

> Maybe I don't understand the question, but I think this is easy isn't it?

It's very easy to check a possible solution, but harder to construct one - is there a valid value for `fractions(Fraction(34649, 49155), Fraction(76540830616, 108585082655), 973)` for example?  Note that checking each possible denominator is not a solution, as max_denom can be arbitrarily large.  (even O(n) is slow for n>10**300)

I've ended up falling back on a solution that will have some false negatives - requiring that a fraction of max_denominator exist between min_value and max_value, or that min_value == max_value with a legal denominator.